bin/fetch-recent-miner-measurements.js

```diff
@@ -0,0 +1,214 @@
+// dotenv must be imported before importing anything else
+import 'dotenv/config'
+
+import { Point } from '@influxdata/influxdb-client'
+import * as Sentry from '@sentry/node'
+import createDebug from 'debug'
+import fs from 'node:fs'
+import { mkdir, readFile, writeFile } from 'node:fs/promises'
+import os from 'node:os'
+import path from 'node:path'
+import pMap from 'p-map'
+import { IE_CONTRACT_ADDRESS } from '../lib/config.js'
+import { createMeridianContract } from '../lib/ie-contract.js'
+import { fetchMeasurements, preprocess } from '../lib/preprocess.js'
+import { RoundData } from '../lib/round.js'
+
+const {
+  STORE_ALL = false
+} = process.env
+
+Sentry.init({
+  dsn: 'https://d0651617f9690c7e9421ab9c949d67a4@o1408530.ingest.sentry.io/4505906069766144',
+  environment: process.env.SENTRY_ENVIRONMENT || 'dry-run',
+  // Performance Monitoring
+  tracesSampleRate: 0.1 // Capture 10% of the transactions
+})
+
+const debug = createDebug('spark:bin')
+
+const cacheDir = path.resolve('.cache')
+await mkdir(cacheDir, { recursive: true })
+
+const [nodePath, selfPath, ...args] = process.argv
+if (args.length === 0 || !args[0].startsWith('0x')) {
+  args.unshift(IE_CONTRACT_ADDRESS)
+}
+const [contractAddress, minerId, blocksToQuery] = args
+
+const USAGE = `
+Usage:
+  ${nodePath} ${selfPath} [contract-address] minerId [blocksToQuery]
+`
+
+if (!contractAddress) {
+  console.error('Missing required argument: contractAddress')
+  console.error(USAGE)
+  process.exit(1)
+}
+
+if (!minerId) {
+  console.error('Missing required argument: minerId')
+  console.error(USAGE)
+  process.exit(1)
+}
+
+if (!minerId.startsWith('f0')) {
+  console.warn('Warning: miner id %s does not start with "f0", is it a valid miner address?', minerId)
+}
+
+console.error('Querying the chain for recent MeasurementsAdded events')
+const measurementCids = await getRecentMeasurementsAddedEvents(contractAddress, blocksToQuery)
+console.error(' → found %s events', measurementCids.length)
+
+const ALL_MEASUREMENTS_FILE = 'measurements-all.ndjson'
+const MINER_DATA_FILE = `measurements-${minerId}.ndjson`
+const MINER_SUMMARY_FILE = `measurements-${minerId}.txt`
+
+const allMeasurementsWriter = STORE_ALL
+  ? fs.createWriteStream(ALL_MEASUREMENTS_FILE)
+  : undefined
+
+const minerDataWriter = fs.createWriteStream(MINER_DATA_FILE)
+const minerSummaryWriter = fs.createWriteStream(MINER_SUMMARY_FILE)
+minerSummaryWriter.write(formatHeader() + '\n')
+
+const abortController = new AbortController()
+const signal = abortController.signal
+process.on('SIGINT', () => abortController.abort(new Error('interrupted')))
+
+try {
+  await pMap(measurementCids, fetchAndProcess, { concurrency: os.cpus().length })
```

Comment:
```suggestion
  await pMap(measurementCids, fetchAndProcess, { concurrency: os.availableParallelism() })
```

> os.cpus().length should not be used to calculate the amount of parallelism available to an application. Use [os.availableParallelism()](https://nodejs.org/api/os.html#osavailableparallelism) for this purpose.

https://nodejs.org/api/os.html#oscpus

Reply:
Thank you for educating me about these two APIs, I learned something new 😄